File: Main/src/main/java/Objects/Pipeline.java
package Objects;
import java.util.Arrays;
import java.util.List;

public class Pipeline {
    public int exposure = 50;
    public int brightness = 50;
    public String orientation = "Normal";
    public List<Integer> hue = Arrays.asList(0,100);
    public List<Integer> saturation = Arrays.asList(0,100);
    public List<Integer> value = Arrays.asList(0,100);
    public boolean erode = false;
    public boolean dilate = false;
    public List<Integer> area = Arrays.asList(0,100);
    public List<Integer> ratio = Arrays.asList(0,20);
    public List<Integer> extent = Arrays.asList(0,100);
    public boolean is_binary = false;
    public String sort_mode = "Largest";
    public String target_group = "Single";
    public String target_intersection = "Largest";
    public double M = 1;
    public double B = 0;
}

File: chameleon-client/src/store/modules/pipeline.js
export default {
    state: {
        exposure: 0,
        brightness: 0,
        gain: 0,
        rotationMode: 0,
        hue: [0, 15],
        saturation: [0, 15],
        value: [0, 25],
        erode: false,
        dilate: false,
        area: [0, 12],
        ratio: [0, 12],
        extent: [0, 12],
        speckle: 5,
        targetGrouping: 0,
        targetIntersection: 0,
        sortMode: 0,
        multiple: false,
        isBinary: 0,
        calibrationMode: 0,
        videoModeIndex: 0,
        streamDivisor: 0,
        is3D: false,
        targetRegion: 0,
        targetOrientation: 1
    },
    mutations: {
        isBinary: (state, value) => {
            console.log(value)
            state.isBinary = value
        },
        mutatePipeline: (state, {key, value}) => {
            // console.log(`key:${key}, value: ${value}`)
            this.set(state, key, value)
        }

    },
    actions: {},
    getters: {
        pipeline: state => {
            return state
        }
    }
};